# DP: Turn on -D_FORTIFY_SOURCE=2 by default for C, C++, ObjC, ObjC++.

---
 gcc/doc/invoke.texi |    6 ++++++
 gcc/gcc.c           |    1 +
 2 files changed, 7 insertions(+), 0 deletions(-)

--- a/src/gcc/gcc.c
+++ b/src/gcc/gcc.c
@@ -810,6 +810,7 @@ static const char *cpp_unique_options =
  %{H} %C %{D*&U*&A*} %{i*} %Z %i\
  %{fmudflap:-D_MUDFLAP -include mf-runtime.h}\
  %{fmudflapth:-D_MUDFLAP -D_MUDFLAPTH -include mf-runtime.h}\
+ %{!D_FORTIFY_SOURCE:%{!D_FORTIFY_SOURCE=*:%{!U_FORTIFY_SOURCE:-D_FORTIFY_SOURCE=2}}}\
  %{E|M|MM:%W{o*}}";
 
 /* This contains cpp options which are common with cc1_options and are passed
